src/features/movies/MovieListPage/index.js

```diff
@@ -1,15 +1,56 @@
 import React, { useEffect } from "react";
-const { useDispatch } = require("react-redux");
-const { fetchMoviesByName } = require("../moviesSlice");
+import Header from "../../../common/Header";
+import Wrapper from "../../../common/Wrapper";
+import { getGenreName } from "../getGenreName";
+import { MoviesContainer } from "../MoviesContainer";
+import MovieTile from "../MovieTile";
+const { useDispatch, useSelector } = require("react-redux");
+const { fetchMoviesByQuery, selectMovies, selectLoading, selectGenres, fetchGenres, selectSearchQuery } = require("../moviesSlice");
 
 const MovieListPage = () => {
-  const dispatch = useDispatch();
+ const dispatch = useDispatch();
 
   useEffect(() => {
-    dispatch(fetchMoviesByName("mulan"));
+    dispatch(fetchGenres());
+    dispatch(fetchMoviesByQuery("frozen"));
   }, [dispatch]);
 
-  return <h1>Cześć !</h1>;
+  const loading = useSelector(selectLoading);
+  const movies = useSelector(selectMovies);
+  const genres = useSelector(selectGenres);
+  const query = useSelector(selectSearchQuery);
+
+  const mapGenreIds = (genreIds, genres) => {
+    return genreIds.map((genreId) => {
+      return getGenreName(genreId, genres);
+    });
+  };
+
+  const generateMovies = (movies) => {
+    return movies.map((movie) => {
+      const genreNames = mapGenreIds(movie.genre_ids, genres);
+      return (
+        <MovieTile
+          key={movie.id}
+          genreNames={genreNames}
+          movie={movie}
+        ></MovieTile>
+      );
+    });
+  };
+
+  if (!loading) {
+    return (
+      <div className="App">
+        <Wrapper>
+        <Header>Search results for "{query}"</Header>
+          <MoviesContainer>{generateMovies(movies.results)}</MoviesContainer>
+        </Wrapper>
+      </div>
+    );
+  } else {
+    return <div className="App">Trwa ładowanie...</div>;
+  }
 };
 
 export default MovieListPage;
```

Comment:
Ogólnie, to ten komponent powinien być w przyszłości połączony z komponentem MoviesPage, bo tak na prawdę jego struktura jest identyczna, zmienia się tylko treść, czyli te film, ale na razie, żeby sprawdzić czy query działa super, że go dodałeś.

Reply:
no to jeżeli już będziemy miec podpiętą nawigację to po prostu zrobimy warunek dotyczący inputu nawigacji